### PR TITLE
Write documentation guidelines document

### PR DESCRIPTION
**Pull request by davidkline-ms**
Write documentation guidelines file based on the v.Next Documentation proposal (#1814). Add steps (and links to the Coding/Documentation guidelines) in the Implementation section of CONTRIBUTING,md to improve process clarity and visibility.

- Fixes: #1893.

### CODE REVIEW

File: DocumentationGuide.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+PME documentation is **required** for a Pull Request to be approved.
+
+As part of a PME summary block, the meaning and purpose of parameters and returned data is required.
+
+```
+        /// <summary>
+        /// Sets the cached native cutoff frequency of the attached low pass filter.
+        /// </summary>
+        /// <param name="frequency">The new low pass filter cutoff frequency.</param>
+        /// <returns>The new cutoff frequency value.</remarks>
```
[the remark this answers is not in the document]
Comment (StephenHodgson) [reply DocumentationGuide.md]: or it's missing the opening `<remarks>` your choice.

File: DocumentationGuide.md [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+### Serialized Fields
+It is a good practice to use Unity's Tooltip attribute to provide runtime documentation for a script's fields in the Inspector.
+
+So that configuration options are included in the API documentation, scripts are required to include *at least* the tooltip contents in a summary block.
+
+```
+        /// <summary>
+        /// The quality level of the simulated audio source (ex: AM radio).
+        /// </summary>
+        [Tooltip("The quality level of the simulated audio source.")]
```
Comment (StephenHodgson) [inline DocumentationGuide.md]: I wish there was an easy way to just add the tooltip as the documentation.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -28,9 +28,11 @@ Note:  If you wish to work on something that already exists on our backlog, you
...
-7. Ensure you update the [README](README.md) with additional documentation as needed.
-8. Also update the [MixedRealityToolkit-Unity wiki](https://github.com/Microsoft/MixedRealityToolkit-Unity/wiki) if you think it will be useful for other developers.
+6. Follow the [Coding Guidelines](/CodingGuidelines.md).
+7. Ensure the code and feature(s) are documented as describred in the [Documentation Guidelines](/DocumentationGuidelines.md).
+8. Ensure the code is [WACK compliant](https://developer.microsoft.com/en-us/windows/develop/app-certification-kit). To do this, generate a Visual Studio solution, right click on project; Store -> Create App Packages. Follow the prompts and run WACK tests. Make sure they all succeed.
+9. Ensure you update the [README](README.md) with additional documentation as needed.
+10. Also update the [MixedRealityToolkit-Unity wiki](https://github.com/Microsoft/MixedRealityToolkit-Unity/wiki) if you think it will be useful for other developers.
 
 # Rebase commits
```
[the remark this answers is not in the document]
Comment (davidkline-ms) [reply CONTRIBUTING.md]: The rebasing comment predates my changes :)

File: DocumentationGuide.md [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+
+In addition to the above, the code should be well commented to allow for maintenance, bug fixes and ease of customization. 
+
+### Class, Struct, Enum Summary Blocks
+If a class, struct or enum is being added to the MRTK, it's purpose must be described. This is to take the form of a summary block above the class.
+
+```
+    /// <summary>
+    /// AudioOccluder implements IAudioInfluencer to provide an occlusion effect.
+    /// </summary><< >>
```
[the remark this answers is not in the document]
Comment (davidkline-ms) [reply DocumentationGuide.md]: Thanks! You caught one of my "write something here" annotations

File: DocumentationGuide.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (davidkline-ms) [reply DocumentationGuide.md]: </remarks> should be </returns>

Good catch!

File: DocumentationGuide.md (hunk 2 above)
Comment (davidkline-ms) [reply DocumentationGuide.md]: Yeah. Me too.

File: CONTRIBUTING.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (StephenHodgson) [reply CONTRIBUTING.md]: No we didn't.  Yeah I think rebasing can cause more headaches than it's worth. As long as there's not conflicts we should be okay.

File: DocumentationGuide.md [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+    /// <remarks>
+    /// Ensure that all sound emitting objects have an attached AudioInfluencerController. 
+    /// Failing to do so will result in the desired effect not being applied to the sound.
+    /// </remarks>
+```
+
+Pull Requests submitted without summaries for classes, strutures or enums will not be approved.
+
+### Property, Method, Event Summary Blocks
+As with classes, structures and enums, PMEs (and fields) are to be documented with summary blocks, regardless of code visibility (public, private, protected and internal). The documentation generation tool is responsible for filtering out and publishing only the public and protected features.
```
Comment (SimonDarksideJ) [inline DocumentationGuide.md]: Explain what a PME is before using the acronym. Always explain terms before their use.

Comment (davidkline-ms) [reply DocumentationGuide.md]: Thanks. That was a casualty of a change I made at the last minute....

**Pull request merged**